### PR TITLE
Removes use of `is_active` for ANS queries

### PR DESCRIPTION
**Pull request by kaw2k**
### Description
[An issue was raised for the ANS api](https://github.com/aptos-labs/aptos-names/pull/313). This PR brings the logic from that fix the the SDK. 

### Test Plan
See the test plan for the linked PR.

### CODE REVIEW

File: src/internal/ans.ts [hunk 1]
```diff
@@ -68,6 +68,28 @@ export function isValidANSName(name: string): { domainName: string; subdomainNam
   };
 }
 
+enum SubdomainExpirationPolicy {
+  Independent = 0,
+  FollowsDomain = 1,
+}
+export function isActiveANSName(name: GetANSNameResponse[0]): boolean {
```
Comment (0xmaayan) [inline src/internal/ans.ts:75]: can we get a test for this function?

File: src/internal/ans.ts [hunk 2]
```diff
@@ -68,6 +68,28 @@ export function isValidANSName(name: string): { domainName: string; subdomainNam
   };
 }
 
+enum SubdomainExpirationPolicy {
```
Comment (0xmaayan) [inline src/internal/ans.ts:71]:
```suggestion
export enum SubdomainExpirationPolicy {
```

File: src/internal/ans.ts (hunk 1 above)
Comment (0xmaayan) [inline src/internal/ans.ts:75]: lets add doc string comments for this function and what it is for

File: src/internal/ans.ts [hunk 3]
```diff
@@ -365,7 +386,7 @@ export async function getName(args: {
     res = sanitizeANSName(res);
   }
 
-  return res;
+  return isActiveANSName(res) ? res : undefined;
```
Comment (0xmaayan) [inline src/internal/ans.ts:389]: dow e need to update this function test accordingly?

Comment (banool) [reply src/internal/ans.ts:389]: We deploy ANS to the localnet as part of the E2E test suite, maybe now is the time to switch the test to use that.

**Review by 0xmaayan — APPROVED** [review]
LGTM

**Pull request merged**